### PR TITLE
Feature/bancapp govgr client

### PR DESCRIPTION
**Pull request by NikosDevPhp**

### CODE REVIEW

**Comment by NikosDevPhp** [comment]
check package version when merged

File: src/Indice.Features.GovGr/Http/BancappAuthorizationMessageHandler.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+{
+    private const string TenantNameProd = "myhands2aade";
+    private const string TenantNameStage = "myhands2testaade";
+    private const int TokenExpirationInSeconds = 30;
+    
+    private readonly GovGrOptions.BancappOptions _options;
+    private readonly IDistributedCache _cache;
+    private readonly HttpClient _httpClient;
+    private string Scope => $"openid {_options.ClientId} offline_access";
+    private string AccessTokenKey => $"{Scope}|{_options.ClientId}|access-token";
```
Comment (cleftheris) [inline src/Indice.Features.GovGr/Http/BancappAuthorizationMessageHandler.cs:22]: Prefix the key with something specific to Bancapp. "Bancapp|....."

File: src/Indice.Features.GovGr/Configuration/GovGrOptions.cs [hunk 2]
```diff
@@ -77,4 +79,33 @@ public class BusinessRegistryOptions
         /// <summary>Callers Fiscal Code</summary>
         public string CallersFiscalCode { get; set; }
     }
-}
+
+    /// <summary>Bancapp settings</summary>
+    public class BancappOptions
+    {
+        /// <summary>Represents the environment. Valid options are <em>production</em>, <em>staging</em>, <em>development</em> &amp; <em>mock</em>.Defaults to <b>stage</b>. </summary>
+        public string Environment { get; set; }
```
Comment (cleftheris) [inline src/Indice.Features.GovGr/Configuration/GovGrOptions.cs:87]: Consider changing this to an Enum of fixed environment names

File: src/Indice.Features.GovGr/GovGrBancappClient.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        GovGrOptions.BancappOptions settings) {
+        _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+        _settings = settings ?? throw new ArgumentNullException(nameof(settings));
+    }
+    
+    private string FunctionUrl => _settings.IsProduction ? FunctionFqdnProd : FunctionFqdnStage;
+    
+    
+    /// <summary>Upload data to eGov Bancapp</summary>
+    public async Task<BancappGCloudUploadResponse> UploadFile(byte[] fileBytes, string fileName) {
```
Comment (cleftheris) [inline src/Indice.Features.GovGr/GovGrBancappClient.cs:28]: Use a stream as the base implementation. Then wrap it around an extension method.

File: src/Indice.Features.GovGr/Configuration/GovGrOptions.cs (hunk 2 above)
Comment (NikosDevPhp) [reply src/Indice.Features.GovGr/Configuration/GovGrOptions.cs:87]: Will do in the next iteration

**Pull request merged**